### PR TITLE
update files for focal cn prep module to use gencode v38 (1/2)

### PR DESCRIPTION
**Pull request by sangeetashukla**
### Purpose/implementation Section
Update the module run file to download Gencode v38 file and run the module.

#### What was your approach?
Use [this link](ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_38/gencode.v38.annotation.gtf.gz) to download Gencode v38 file. 

#### What GitHub issue does your pull request address?
[Issue 320](https://github.com/PediatricOpenTargets/ticket-tracker/issues/320)

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.
On running the entire module, two new files were added to the `results` directory. Please review them.
```
consensus_seg_annotated_cn_x_and_y.tsv.gz
consensus_seg_annotated_cn_autosomes.tsv.gz
```

#### Which areas should receive a particularly close look?
Please review the downloaded gencode file version and link it was downloaded from.

#### Is there anything that you want to discuss further?
No

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What types of results are included (e.g., table, figure)?
Three new files are included
```
gencode.v38.annotation.gtf.gz
consensus_seg_annotated_cn_x_and_y.tsv.gz
consensus_seg_annotated_cn_autosomes.tsv.gz
```

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by sangeetashukla** [comment]
This branch has now been merged into another [PR](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/182)

This PR and [ticket](https://github.com/PediatricOpenTargets/ticket-tracker/issues/320) can be closed.

**Comment by jharenza** [comment]
> This branch has now been merged into another [PR](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/182)
> 
> This PR and [ticket](https://github.com/PediatricOpenTargets/ticket-tracker/issues/320) can be closed.

Thanks for this, but we will still review these separately. The intent is to stagger the PRs, so different aspects are reviewed separately. This should remain open and reviewed.

File: analyses/focal-cn-file-preparation/01-add-ploidy-cnvkit.Rmd [hunk 1]
```diff
@@ -122,7 +122,7 @@ add_xy_df <- add_ploidy_df %>%
     germline_sex_estimate == "Male" & copy.num == (0.5 * tumor_ploidy) ~ "neutral",
     # there are some instances where there are chromosome Y segments are in
     # samples where the germline_sex_estimate is Female
-    chrom == "chrY" & germline_sex_estimate == "Female" & copy.num > 0 ~ "unknown",
+    chrom == "chrY" & germline_sex_estimate == "Female" & copy.num > 0 ~ as.character(NA),
```
Comment (jharenza) [inline analyses/focal-cn-file-preparation/01-add-ploidy-cnvkit.Rmd:125]: Can you revert this for this PR, as it is the main code change in #182

File: analyses/focal-cn-file-preparation/run-prepare-cn.sh [hunk 2]
```diff
@@ -19,92 +19,80 @@ script_directory="$(perl -e 'use File::Basename;
...
 
-# Run snakemake script implementing `bedtools coverage` for each sample bed file in
-# `scratch/cytoband_status` -- these files are generated in
-# `02-add-ploidy-consensus.Rmd`
-# currently runs 10 jobs in parallel, which should be fine for most implementations
-snakemake -j 10 --snakefile run-bedtools.snakemake
-
-# Determine the dominant status for each chromosome arm and compare with GISTIC's arm status calls
-Rscript --vanilla -e "rmarkdown::render('03-add-cytoband-status-consensus.Rmd', clean = TRUE)"
-
```
Comment (jharenza) [inline analyses/focal-cn-file-preparation/run-prepare-cn.sh:41]: Do you know why this code is missing now?

File: analyses/focal-cn-file-preparation/run-prepare-cn.sh [hunk 3]
```diff
@@ -19,92 +19,80 @@ script_directory="$(perl -e 'use File::Basename;
...
-      --goi_list $goi_file \
-      --filename_lead "consensus_seg_annotated_cn"_${chromosome_type}_${strategy}
-  done
-done
+--cnv_file ${scratch_dir}/consensus_seg_with_status.tsv \
+--gtf_file $gtf_file \
+--metadata $histologies_file \
+--filename_lead "consensus_seg_annotated_cn" \
+--seg
+
```
Comment (jharenza) [inline analyses/focal-cn-file-preparation/run-prepare-cn.sh:50]: Looks like this was also deleted

**Review by jharenza — CHANGES_REQUESTED** [review]
see comments

File: analyses/focal-cn-file-preparation/01-add-ploidy-cnvkit.Rmd (hunk 1 above)
Comment (sangeetashukla) [reply analyses/focal-cn-file-preparation/01-add-ploidy-cnvkit.Rmd:125]: @jharenza the branch in [current PR](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/181) is now merged into [#182](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/182)\\

We can ignore this PR and merge only the other one

File: analyses/focal-cn-file-preparation/run-prepare-cn.sh (hunk 2 above)
Comment (sangeetashukla) [reply analyses/focal-cn-file-preparation/run-prepare-cn.sh:41]: @runjin326 confirmed on slack that we want to skip this code for generating v11 files.
I will add her as a reviewer for [#182](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/182)

File: analyses/focal-cn-file-preparation/run-prepare-cn.sh (hunk 3 above)
Comment (sangeetashukla) [reply analyses/focal-cn-file-preparation/run-prepare-cn.sh:50]: @jharenza Sure, I will look into this.

**Comment by jharenza** [comment]
Going to close since we can't update base branches on remote

**Pull request closed without merging**